worlds/sc2/mission_order/options.py

```diff
@@ -0,0 +1,222 @@
+                "limit": 0,
+                "required": False,
+                "unlock_count": -1,
+                "unlock_specific": [],
+                "mission_pool": ["all missions"],
+                "min_difficulty": "relative",
+                "max_difficulty": "relative",
+            },
+            "Default Layout": {
+                "type": "grid",
+                "size": 9,
+                "limit": 3,
+            },
+        },
+    }
+    schema = Schema({
+        # Campaigns
+        str: {
+            "order": int,
+            "unlock_count": IntNegOne,
+            "unlock_specific": [str],
+            "required": bool,
+            "min_difficulty": Difficulty,
+            "max_difficulty": Difficulty,
+            # Layouts
+            str: {
+                # Type options
+                "type": lambda val: issubclass(val, LayoutType),
+                "order": int,
+                "size": IntOne,
+                "limit": IntZero,
+                # Link options
+                "required": bool,
+                "unlock_count": IntNegOne,
+                "unlock_specific": [str],
+                # Mission pool options
+                "mission_pool": {int},
+                "min_difficulty": Difficulty,
+                "max_difficulty": Difficulty,
+                # Mission slots
+                Optional(int): {
+                    Optional("mission"): str,
+                    Optional("required"): bool,
+                    Optional("entrance"): bool,
+                    Optional("exit"): bool,
+                    Optional("empty"): bool,
+                    Optional("next"): [int],
+                    Optional("unlock_count"): IntZero,
+                    Optional("unlock_specific"): [str],
+                    Optional("mission_pool"): {int},
+                    Optional("difficulty"): Difficulty,
+                },
+            },
+        }
+    })
+    
+    def __init__(self, value: typing.Dict[str, typing.Dict[str, typing.Any]]):
+        # Make sure all the globals are filled
+        self.value = dict()
+        if value == self.default: # If this option is default, it shouldn't mess with its own values
+            value = copy.deepcopy(self.default)
+        for campaign in value:
+            self.value[campaign] = dict()
+            # Check if this campaign has a global layout
+            global_dict = dict()
+            for name in value[campaign]:
+                if name.lower() == GLOBAL_ENTRY:
+                    global_dict = value[campaign].pop(name, dict())
+                    break
+            self.value[campaign] = {key: value for (key, value) in self.default["Default Campaign"].items() if type(value) != dict}
+            self.value[campaign].update(value[campaign])
+            _resolve_special_options(self.value[campaign])
+            for layout in value[campaign]:
+                if type(value[campaign][layout]) != dict:
+                    continue
```

Comment:
I only just remembered to reply to this, but this wouldn't be a format error, the value would just be an option instead of a layout. This ties back to our lengthy conversations about having a list of layouts versus declaring layouts by name, so I'll just mark this resolved, and if we ever do decide to change to your approach, this is guaranteed to change too.